### ISSUE TITLE
Bedtime Mode not being synced on Pixel Watch

### ISSUE THREAD

**Issue by Dave247** [dreadedlama/dnd-sync#1]
(Follow on from https://github.com/rhaeus/dnd-sync/issues/11)

I tried version 2.2 which does seem to work a lot better then I remember the app used to. However, I don't think Bedtime mode still gets set correctly on the synced device, mine being a Pixel Watch and Pixel 7 Pro.

When enabling Bedtime mode using the quick settings tile on either the watch or phone, Do not Disturb is correctly turned on, but the Bedtime mode tile on the synced device doesn't change to indicate that the mode is turned on.

Honestly though as the mode isn't explained all that well, I'm not sure exactly what difference it even makes, or if it even matters.

**Comment by DreadedLama** [comment]
The bedtime mode sync is a one-way sync and not a 2-way sync like DND, it works only from mobile to watch, not the other way around.
I have tested it on a Galaxy watch 4, and not on a pixel watch as I don't have one.

Can you try turning on bedtime mode on your mobile and check if it syncs to your Pixel watch?

**Comment by Dave247** [comment]
Ah hmm, good to know, that narrows it down what I should be testing and looking for.

Unfortunately though the result is the same, turning on Bedtime mode on my Pixel watch only turns on DND on my phone, but not Bedtime mode.

What phone are your trying it on with your Galaxy Watch? As maybe it might not just be something with the watch, but also because I have a Pixel 7, where Digital Wellbeing is integrated pretty heavily into the stock ROM. On top of that, I'm also using the beta version in case that adds any difference. According to the appinfo, I have version `1.6.551070056.beta (529558)` of Digital Wellbeing installed.

**Comment by DreadedLama** [comment]
I have a Pixel 7 paired to my Galaxy watch. I'm too on Android 14 Beta.

The watch is supposed to turn to bedtime mode if I turn my phone to bedtime mode and not the other way around. BetimeMode feature is a one way sync only(from mobile to watch).

Once you turn bedtime mode on mobile, it sends an intent to watch, and upon receiving it the watch app changes the SECURE_SETTING "setting_bedtime_mode_running_state" setting value from 0 to 1. Galaxy watch uses this secure setting value to change its bedtime mode, not sure if the Pixel watch has the same setting for bedtime mode. Maybe you can test it and confirm if it works in the desired way.

On the contrary, changing the bedtime mode on the watch does not send any intent to the mobile app. It is because I was unable to find any SECURE_SETTING on mobile which can be overwritten to change the bedtime mode. The one way that I can think of is to use accessibility service for it on mobile(to mimick input swipes), which I personally don't want to.

Though there are workaround to it, like just turning the mobile to grayscale mode, which I think can be done without mimicking input swipes (you can find this option in Accessibility->Colour Correction), which in a sense is what bedtime mode does, it turns your phone black and white. But this approach also has a few issues from an implementation perspective. I'm still looking to a viable approach for this (if possible)

**Comment by Dave247** [comment]
Oh sorry, I misunderstood and had it the wrong way around. The result is still the same though, I don't see the little bedtime mode tile on my watch light up when toggling it on my phone. It only toggles DND.

I did grant the secure settings permission which shows as enabled on the app on the watch though, as well as of course toggling the setting for bedtime mode on. I didn't get any error when running the command over adb so I assumed for it to have worked.

How would I go about looking for the setting in the rom on my watch? While I don't know the ins and outs of Android in intricate detail, I know enough to be able to follow instructions and find the answer out for myself, as long as I know what I should be doing.

**Comment by DreadedLama** [comment]
I used autowear for this, you can install autowear on both mobile and watch from Play Store, grant it necessary permissions. I guess you get 7 days of free access before you need to buy it (not sure though as I have premium version). 
Grant the watch app adb WRITE_SECURE_SETTINGS permission using the following command- 
"adb shell pm grant com.joaomgcd.autowear android.permission.WRITE_SECURE_SETTINGS"

Once you have installed the apps and granted the permissions, these are the steps - 
1) In the mobile app, there is an option known as "Secure Settings. Open it and click on "+"
2) Click "Automatic Setup"
3) Now change some setting on the watch, in this case, set the watch to Bedtime mode.
4) Click "ok" on mobile. 
5) On the mobile app it will show the changed secure settings which have been modified on the watch.

You can refer to this video for reference - 
https://youtu.be/Rej1y5r34Bw?t=216

Once you get all the settings, can you please post a screenshot of them here for reference?

Do let me know if you run into any issues.

**Comment by Dave247** [comment]
Oh wow, that's amazing. I was fully expecting needing to do some complex dump and have to sift through looking for what I needed, but this app is great. I didn't even need to activate the trial to be able to use the secure settings feature. I should spend some more time looking into the app later and see if there is anything I want to do that I could use it for.

Anyway so here is the screenshot. It looks like the flag has a different name on the Pixel Watch, hence why it wasn't working.
It also seems to toggle mobile data, though I don't have the LTE variant so that shouldn't even be a thing for me, but I guess the flag is still there because the rom would be basically the same.
There is also an extra `zen_mode`, but from what I found online about it that's a OnePlus feature, so I'm not sure what it is doing here. Unless what it actually is is Focus mode? As that seems to be the closest comparable thing.
Also I did it several times to be sure, including restarting the watch and trying it again, but the zen mode etag value will always change between those two numbers and nothing else, whatever that even means. 
![Pixel Watch Secure Settings Screenshot ](https://github.com/DreadedLama/dnd-sync/assets/10183015/b4e776fc-e6db-48a7-88e5-c733786e24f4)

**Comment by DreadedLama** [comment]
Zen_mode is for DND ( I see that flag toggle when I turn on DND on my watch), it is there on Galaxy watch too, but directly setting it using secure setting doesn't seem to work for me.  Thanks for the quick update though.
I still have to think about how can I possibly differentiate between the galaxy and pixel watch on the watch app to know which one is which. Maybe I'll just simple set both the Galaxy watch and pixel watch flags and hope it works on both. Will have to try it to know for sure.
 I'm currently busy with some other stuff so will update the app once I get time.

Autowear(Watch automation) is a gem and so is Tasker(mobile automation). The combo together can nearly do anything you wish.

This app is for people who don't want to go into such great depths and just want a few basic features which logically should have been in wearos by default.

**Comment by Dave247** [comment]
Yeah, this really should be an option built into Android itself. I think the idea has been floated and talked about over the years, but nothing ever seems to have come of it sadly.

As for differentiating between the two, you should be able to get the model/code name seeing as it's still just Android right? Or is that not so simple?

And yeah I get being busy, don't worry about it. Just post back in here when you do it, or have any other ideas you'd like me to try and I'll get an email notification for it so I won't miss it.

**Comment by DreadedLama** [comment]
I haven't tried to get model. Will try to see how easy it is. Will update  here on any progress

**Comment by Dave247** [comment]
Also one other thing I just thought of, there's also this app which might make things easier to access certain root API's. Though it would probably require some major rewriting of the code though to use it.
https://shizuku.rikka.app/

**Comment by DreadedLama** [comment]
Shizuku is really cool for mobile but its not so focused for wear os as of now. I only found 1 app till now on wearos which uses shizuku and it was to use a galaxy watch exploit if I remember correctly. Wearos developers don't use it as of now. For now, I don't have any plan to use it because as you mentioned it will require major changes in code.

**Comment by Dave247** [comment]
Yeah I figured it probably wouldn't work on Wear OS, and seeing as most of what you want to do is on the watch side of things, it wouldn't really be of much use. Ah well, there is only so much you can do. At least syncing DND alone is nice without needing to fuss around with another app such as Autowear.

**Comment by WaleedAshraf** [comment]
Hi Both,

I have Nothing Phone with Pixel Watch 2. I have the same issue,
- DND Sync works perfectly.
- Bedtime sync turns on/off DND instead.

Were you able to find a fix for it?

**Comment by DreadedLama** [comment]
HI @Dave247  @WaleedAshraf,
I have done a new release with changes to sync bedtime mode on pixel watches, can you please test it and let me know if it works as expected? 
Please use the latest mobile and wear os apps
[V2.3](https://github.com/DreadedLama/dnd-sync/releases/tag/v2.3)

**Comment by Dave247** [comment]
> HI @Dave247  @WaleedAshraf,
> I have done a new release with changes to sync bedtime mode on pixel watches, can you please test it and let me know if it works as expected? 
> Please use the latest mobile and wear os apps
> [V2.3](https://github.com/DreadedLama/dnd-sync/releases/tag/v2.3) 

Unfortunately as I have given my Pixel watch away to a family member I am unable to test it.

**Comment by WaleedAshraf** [comment]
Thanks @DreadedLama. I'll test it in coming days and will update here.

**Comment by d3ighty** [comment]
i just tested it. Doesnt work for me. Still just puts on dnd mode and not sleepmode. Tested on op12 and pw1

**Comment by arjunambati** [comment]
I tested V2.3 to sync (one way) bedtime mode from S24 Ultra to Pixel Watch 2. It didn't work for me. DND works perfectly.

**Comment by alontraut** [comment]
Hello,

I just test v2.4 and see that bedtime mode still doesn't work. I have a Pixel Watch 4 with a Samsung Galaxy S26 Ultra. Is there any further development happening on this issue and is there anything I can do to assist?

**Comment by DreadedLama** [comment]
@alontraut  I’m not actively working on the project at the moment, but I’ll try to pick this up when I get some time (I can’t promise a timeline right now). Since I don’t have a Pixel Watch to test with, I’ll likely need your help with a few things once I start investigating.

I’ll follow up in this thread with specific steps when I’m able to work on it again.

**Comment by dreadedlama** [comment]
> Hello,
> 
> I just test v2.4 and see that bedtime mode still doesn't work. I have a Pixel Watch 4 with a Samsung Galaxy S26 Ultra. Is there any further development happening on this issue and is there anything I can do to assist?


Hi @alontraut, I’ve just released a new version of the app. Could you give it a try and let me know if it works better for you?

Also, could you follow the steps I mentioned above to @Dave247 and share a screenshot from the AutoWear app showing which settings are being changed? That would help me understand what’s happening on the watch and investigate the issue further.

Thanks!

**Comment by Dave247** [comment]
This brings back memories. Didn't expect to see a notification for it pop up, but it's ok though as I had an idea.

I've been fortunate that since posting the original issue I moved on to sticking with the same brand/ecosystem for watch and phone. First the Galaxy Watch 6 and the S21 and later Z Pro Fold 5, then the Pixel Watch 3 and Pixel Pro Fold 9 and later 10.

I admit I've been taking the syncing for granted and missed it when I had to send my Pixel phone off for RMA and had to swap back to the Samsung phone until I received a replacement (which I had intended to sell but never got around to it, so it came in handy).


I realised when seeing the mention that I could try pairing the pixel watch with the Samsung phone and vice versa to see if the syncing works with the app. Even though I don't need the app for myself anymore, It's the least I can do for everyone else seeing as I was the one who originally started the issue.


Also I might get to finally understand what the flag is that I can use as an action in Tasker for Bedtime mode, as some time ago I started wearing my watch to bed for sleep tracking which while the watch can automatically turn on Bedtime mode when it detects you are asleep (and then subsequently sync that to the phone), I still continue to use Sleep as Android and have always wanted to have it so when I manually start sleep tracking in the app, it would also turn on Bedtime mode on the phone (which would then also get synced to the watch).

**Comment by dreadedlama** [comment]
That would be great @Dave247, thank you!

I wasn’t actively working on the app for a while since it was doing what I needed, but recently I decided to take another look at it and see if I could finally resolve the Pixel Watch syncing issue.
I’ve noticed the same Pixel Watch issue being reported across multiple forks of DND-Sync, so I thought I’d give it one more serious attempt and hopefully get it working properly for everyone.

Since I only have a Samsung watch myself, I unfortunately can’t reproduce the Pixel Watch behavior directly. Being able to pair the Pixel Watch with the Samsung phone (and vice versa) and capture the changes in flag would be extremely helpful.

The Tasker/Bedtime Mode part is also interesting. If we can identify exactly which settings/flags change when Bedtime Mode is enabled or disabled, that may give us a much better understanding of how Google is handling the sync and potentially help with both the DND-Sync implementation and your Sleep as Android use case.

I’d really appreciate your help with this. If you can capture the AutoWear screenshots/logs for the different scenarios, I’ll dig into them and see what I can figure out.

**Comment by Dave247** [comment]
Here's the detected differences from my Pixel Watch 4 on my Pro Fold 10.
![Screenshot_20260905-230539.png](https://github.com/user-attachments/assets/d08d653f-5a2c-43a3-a920-9ded123f4679)


And here it is on the Z Fold 5
![Screenshot_20260905_235012_AutoWear.jpg](https://github.com/user-attachments/assets/8ed5c51a-87e3-4f3f-8afd-7114019c159a)


(Side note, I'm thankful that Google made transfering a watch to a different phone super simple and not require having to factory reset the watch anymore as that was a absolute pain.)


I also tried version 2.5 on my watch and Samsung phone, but unfortunately toggling Bedtime (or Sleep as it is called on Samsung) still acts the same where it is only DND mode is toggled on the other device.


Possibly unrelated but when I was using Autowear to capture the flags an error popped up which I have sent off to Joao, but I feel like it was doing the same thing even back three years ago so it may not really matter.


Still with how convoluted this whole thing is it could be anything. While looking I happed to stumble across [this old feature request for Tasker that I participated in related to the addition of Bedtime mode](https://tasker.helprace.com/i1894-listener-for-new-modes-presets), so I don't know if there is anyone who even understands how it works aside from Samsung who in their infinite wisdom have made some changes as how to modes operate as part of their routines.

**Comment by dreadedlama** [comment]
Thanks for sharing the info! I’ll go through it, and I may ask you to test a pre-release build later to see if it works properly with the Pixel Watch.


> I also tried version 2.5 on my watch and Samsung phone, but unfortunately toggling Bedtime (or Sleep as it is called on Samsung) still acts the same where it is only DND mode is toggled on the other device.

Going through what you mentioned, I think I may have found why the app doesn’t seem to work properly with Samsung phones. The mobile app relies on Google’s Digital Wellbeing app (`com.google.android.apps.wellbeing`) to detect the Bedtime mode state. Since Samsung uses its own implementation instead of Google’s Digital Wellbeing, the mobile app never gets the information about whether Bedtime mode is enabled.

Basically, the mobile app is checking for a notification from Google’s Digital Wellbeing app to determine the Bedtime mode state.

I have a Samsung phone lying around, so I’ll dig into it and see if I can find something useful there. Maybe there’s a way to detect Samsung’s Sleep mode or trigger the Bedtime sync through a different mechanism. That could potentially help get the sync working on Samsung devices as well.

**Comment by Dave247** [comment]
> Going through what you mentioned, I think I may have found why the app doesn’t seem to work properly with Samsung phones. The mobile app relies on Google’s Digital Wellbeing app (`com.google.android.apps.wellbeing`) to detect the Bedtime mode state. Since Samsung uses its own implementation instead of Google’s Digital Wellbeing, the mobile app never gets the information about whether Bedtime mode is enabled.
> 
> Basically, the mobile app is checking for a notification from Google’s Digital Wellbeing app to determine the Bedtime mode state.

Great work on taking the time to dig into it, as I'm sure people are going to appreciate it as I haven't come across any people who have put in the time to try and wrap their head around what the differences are between the two, let alone the changes to modes/zen in recent android versions.

I've always found it confusing, as on the surface I thought that when Google first introduced digital wellbeing that it was like a framework, and that Samsung have just made adaptations (as they do with everything). But if that was the case I would have thought that the two should share greater comparability. Instead in a lot of ways it seems like Samsung built their own thing that just acts and operates in the same way (as they also do with a lot of things) when Google's implementation is actually proprietary.

**Comment by michael-r-elp** [comment]
Saw this project, I have the combination pixel watch 5 and samsung galaxy s26 ultra and so far couldn't find any active fork of this project that managed to get bedtime sync to work.

For some weird reason, the only way to get the pixel watch to toggle bedtime mode on is if I do the following:

1. Have sync bedtime mode enabled on the phone
2. Turn DND on, on the watch, have it sync to the phone
3. Disable and reenable DND on the phone, then bedtime mode on the watch will be enabled
4. (Optional) Look confused at your watch and think about why it only does it like this

So it can definitely toggle the setting, but somehow, doesn't

Here's a logcat log of what the app logs on the watch when I do the above and then also try turning DND off and on, on the phone:
```
--------- beginning of main
09-06 19:42:34.428  7738  7738 D DNDNotificationService: interruption filter changed to 2
09-06 19:42:34.561  7738  9126 D DNDNotificationService: message sent to 9333d871: 16668
09-06 19:42:39.293  7738  7738 D DNDNotificationService: interruption filter changed to 1
09-06 19:42:39.321  7738  9141 D DNDNotificationService: message sent to 9333d871: 16670
09-06 19:42:42.448  7738  7738 D DNDNotificationService: interruption filter changed to 2
09-06 19:42:42.645  7738  9169 D DNDNotificationService: message sent to 9333d871: 16671
09-06 19:42:48.908  7738  7738 D DNDNotificationService: interruption filter changed to 1
09-06 19:42:49.097  7738  9207 D DNDNotificationService: message sent to 9333d871: 16672
09-06 19:42:53.648  7738  7738 D DNDNotificationService: interruption filter changed to 2
09-06 19:42:53.728  7738  9266 D DNDNotificationService: message sent to 9333d871: 16673
09-06 19:43:00.625  7738  7738 D DNDNotificationService: interruption filter changed to 1
09-06 19:43:00.648  7738  9298 D DNDNotificationService: message sent to 9333d871: 16674
```
The watch does vibrate twice when I have the vibrate setting enabled, though of course it cannot detect bedtime from samsung digital wellbeing due to it working in a different way and not having a specific notification, though it is possible to setup a custom notification.

I'd really like if we can get it working, though I'm also kinda used to it not working ever since it previously stopped working natively with my old pixel watch and pixel 7 pro I had, similar to the person who opened this issue up.

I do wonder if it not working could be related to it somehow not liking that it tries to toggle both dnd and bedtime, so maybe an option to only toggle bedtime or delay toggling it might work? 
Doing the above steps does seem to lead to it only enabling bedtime and not dnd on the watch and it doesn't seem to always work, mostly end up turning on DND only.

**Comment by michael-r-elp** [comment]
Actually, I think I just found the problem, the watch does not allow having both bedtime and DND turned on, turning one on disables the other.

So we'd need an option to only enable bedtime and not dnd for the watch, disabling bedtime on the watch also seems to have the same effect as disabling dnd and syncs that to the phone

**Comment by Dave247** [comment]
> I'd really like if we can get it working, though I'm also kinda used to it not working ever since it previously stopped working natively with my old pixel watch and pixel 7 pro I had, similar to the person who opened this issue up.

If I recall correctly when I originally opened the issue that was before the update to bring the feature to the original pixel watch, [as it was originally only possible from the pixel watch 2 and up](https://github.com/dreadedlama/dnd-sync/issues/4#issuecomment-1872051800). Are you saying that at some point after the update it was working for you and then inexplicably stopped? That's very strange.

> Actually, I think I just found the problem, the watch does not allow having both bedtime and DND turned on, turning one on disables the other.

Great job spotting that, I don't know how I never thought of it as I totally forgot that DND and Bedtime only one can be enabled at a time, where if you manually turn on either the other is turned off.

**Comment by dreadedlama** [comment]
>Actually, I think I just found the problem, the watch does not allow having both bedtime and DND turned on, turning one on disables the other.

That’s really great that you noticed this, @michael-r-elp! This could very well explain why the Pixel Watch syncing has been so unreliable.

I’ll look into it and try adding a toggle in a pre-release to enable **Bedtime Mode only** on the watch, without syncing DND. Hopefully that gives us a more reliable way to get this working.

Maybe you and @Dave247 could test the pre-release and validate whether it works consistently on your Pixel Watches.

I’ve also noticed that, so far, nobody seems to have fully cracked the Bedtime Mode syncing issue specifically on Pixel Watches, so this is a really useful lead.

**Comment by dreadedlama** [comment]
>The watch does vibrate twice when I have the vibrate setting enabled, though of course it cannot detect bedtime from samsung digital wellbeing due to it working in a different way and not having a specific notification, though it is possible to setup a custom notification.

@michael-r-elp Could you guide me through the custom notification setup? I’m not a Samsung user, so I’m not familiar with this custom notification thing.

I’d like to see exactly what notification we can trigger and whether it could help us detect Bedtime Mode on the watch.

**Comment by michael-r-elp** [comment]
> If I recall correctly when I originally opened the issue that was before the update to bring the feature to the original pixel watch, [as it was originally only possible from the pixel watch 2 and up](https://github.com/dreadedlama/dnd-sync/issues/4#issuecomment-1872051800). Are you saying that at some point after the update it was working for you and then inexplicably stopped? That's very strange.

Yeah it was originally working with the pixel watch app on my original pixel watch but eventually stopped working, probably as the original pixel watch reached its eol.
Though it was also acting a bit weird as well getting the feature initially working I believe. 

Also weirdly enough the toggle for syncing modes still shows up on the pixel app for my Samsung as an option I can toggle but pressing it does nothing.
Kinda weird to have the toggle visible when it's not supposed to be there on non-pixel devices, they do hide the alarm sync feature from the UI on the Google clock app though.

**Comment by michael-r-elp** [comment]
> @michael-r-elp Could you guide me through the custom notification setup? I’m not a Samsung user, so I’m not familiar with this custom notification thing.
> 
> I’d like to see exactly what notification we can trigger and whether it could help us detect Bedtime Mode on the watch.
> 

I actually just checked and it sends a one-off notification but doesn't remove the notification after disabling bedtime mode. 
So you have to manually dismiss it, might not be that useful then after all

**Comment by michael-r-elp** [comment]
Btw, another thing I took a look at using an app called SetEdit.
When enabling bedtime on an samsung phone, it seems to set the following values in the global settings table:

Bedtime Mode Enabled:
```json
{    
    "mode_display_name": "Schlaf",
    "mode_enabled": "1",
    "mode_id": "100",
    "zen_mode": "1"
}
```

Bedtime Mode Disabled:
```json
{
    "mode_display_name": "",
    "mode_enabled": "0",
    "mode_id": "0",
    "zen_mode": "0"
}
```

Though I'm not sure how we'd detect when that setting is changed on the phone, maybe check the value when DND is turned on, since it's usually set to turn DND on the phone on together with bedtime, so we know what we actually need to toggle on the watch.
This way it would be possible to sync DND and bedtime separately to the watch depending on which values we see.

I should add that it is possible to add/delete modes on samsung, so you can technically have bedtime mode on an different id than 100, if you were to delete the original bedtime mode and create a new one I'd guess, but this should still work under the assumption that the factory created modes are left intact.

Edit: `zen_mode` is also enabled but that seems to just indicate dnd

**Comment by dreadedlama** [comment]
> `mode_id`: `100`

Yep, I noticed the `mode_id: 100` value 2 days back as well. Also, `mode_id: 105` corresponds to Theatre Mode.

I’m currently looking into whether we can read these values directly through the app and use them to determine which mode is active, rather than relying only on DND. I’ll update here once I make some progress.

> I should add that it is possible to add/delete modes on Samsung, so you can technically have Bedtime Mode on a different ID than 100 if you delete the original mode and create a new one. This should still work under the assumption that the factory-created modes are left intact.

I wasn’t aware of this part since I’m not a Samsung user. For the time being, I’ll first see if I can reliably read these values from the app. If that works as expected, we can look into handling custom/deleted modes later.

**Comment by michael-r-elp** [comment]
> Yep, I noticed the `mode_id: 100` value 2 days back as well. Also, `mode_id: 105` corresponds to Theatre Mode.
> 
> I’m currently looking into whether we can read these values directly through the app and use them to determine which mode is active, rather than relying only on DND. I’ll update here once I make some progress.
> 

I assume calling the same function in `Settings.Globals` that the wear companion app calls with the corresponding ID here, to check and change bedtime mode on the watch, if we only need to read the value no additional permissions should be needed, I think
Though I believe that what I see in SetEdit isn't the full ID

> I wasn’t aware of this part since I’m not a Samsung user. For the time being, I’ll first see if I can reliably read these values from the app. If that works as expected, we can look into handling custom/deleted modes later.
> 
> 

I did just check what happens when deleting and adding a built in mode back, seems the built-in ones have a hard coded ID, the custom ones on the other hand have a longer random ID, so it seems we're fine here

**Comment by dreadedlama** [comment]
>I wasn’t aware of this part since I’m not a Samsung user. For the time being, I’ll first see if I can reliably read these values from the app. If that works as expected, we can look into handling custom/deleted modes later.

That’s really good to know! This should make the implementation much simpler once we get everything working. I’m currently looking into the implementation and will update here once I make some progress.

Thanks for looking into this and keeping me updated!

**Comment by dreadedlama** [comment]
Hi @Dave247 @michael-r-elp, could you try the builds from this workflow to test the Pixel Watch issue fix?

[GitHub Actions build](https://github.com/dreadedlama/dnd-sync/actions/runs/34246643970)

I’ve added a new toggle in the mobile app:

**“Enable only Bedtime Mode on watch (No DND)”**

Please enable this option and then test whether Bedtime Mode syncs correctly to the Pixel Watch.
You can also try disabling the regular **DND sync** option altogether, just to be extra cautious and isolate the Bedtime Mode behavior.
Please test using **both the mobile and Wear OS apps** and let me know how it behaves

Thanks

**Comment by dreadedlama** [comment]
Oh, I forgot to mention that I haven’t added the Samsung fix yet.

So, if possible, could you validate the Pixel Watch fix using a non-Samsung phone. This would help us confirm that the fix is specific to the watch-side behaviour and isn’t dependent on Samsung specific functionality.

**Comment by michael-r-elp** [comment]
> Oh, I forgot to mention that I haven’t added the Samsung fix yet.
> 
> So, if possible, could you validate the Pixel Watch fix using a non-Samsung phone. This would help us confirm that the fix is specific to the watch-side behaviour and isn’t dependent on Samsung specific functionality.

I unfortunately already packed my old watch for trade-in and it's kinda a painful process to switch the phone they're paired to.

What I did try though is the following:
* Sync DND state to watch: on
* Enable Bedtime mode on DND sync: on
* Sync Bedtime Mode to watch: on
* Enable only Bedtime mode on watch (No DND): on

And that actually works with samsung, though I did have it once toggle DND instead of bedtime but in all other instances I tried it did toggle bedtime on and off.

So if we can get the bedtime detection for samsung working as well now, then I think this app will be perfect.

EDIT: Adding the logcat output from the watch here, when enabling/disabling bedtime:
```
09-08 20:40:18.609  3056 13015 D DNDSyncListenerService: onMessageReceived: MessageEventParcelable[16818,/wear-dnd-sync, size=171]
09-08 20:40:18.609  3056 13015 D DNDSyncListenerService: dndStateWear: 2
09-08 20:40:18.609  3056 13015 D DNDSyncListenerService: currentDndState: 2
09-08 20:40:27.493  3056  3056 D DNDNotificationService: interruption filter changed to 1
09-08 20:40:27.493  3056  3056 D DNDNotificationService: dnd sync is true
09-08 20:40:27.497  3056 13108 D DNDNotificationService: send successful! Receiver node id: 67535a7b (data: 16716)
09-08 20:40:27.829  3056 13132 D DNDSyncListenerService: onMessageReceived: MessageEventParcelable[16819,/wear-dnd-sync, size=171]
09-08 20:40:27.829  3056 13132 D DNDSyncListenerService: dndStateWear: 1
09-08 20:40:27.829  3056 13132 D DNDSyncListenerService: currentDndState: 1
```

**Comment by dreadedlama** [comment]
> What I did try though is the following:
> 
> * Sync DND state to watch: on
> * Enable Bedtime mode on DND sync: on
> * Sync Bedtime Mode to watch: on
> * Enable only Bedtime mode on watch (No DND): on

Could you confirm which watch and mobile you are using for this test?

**Comment by michael-r-elp** [comment]
> Could you confirm which watch and mobile you are using for this test?
> 

Samsung Galaxy S26 Ultra + Pixel Watch 5

**Comment by dreadedlama** [comment]
> What I did try though is the following:
> 
> * Sync DND state to watch: on
> * Enable Bedtime mode on DND sync: on
> * Sync Bedtime Mode to watch: on
> * Enable only Bedtime mode on watch (No DND): on

Apologies if I’m misunderstanding, but are you able to successfully enable Bedtime Mode on the Pixel Watch 5?

If yes, with all four settings mentioned above enabled, does turning on DND on the mobile successfully enable Bedtime Mode on the watch?

And with “Enable Bedtime mode on DND sync” turned off, and “Enable only Bedtime mode on watch (No DND)” also turned off and with the other two settings enabled, are you able to sync DND normally?

**Comment by michael-r-elp** [comment]
> Apologies if I’m misunderstanding, but are you able to successfully enable Bedtime Mode on the Pixel Watch 5?
> 
Yes

> If yes, with all four settings mentioned above enabled, does turning on DND on the mobile successfully enable Bedtime Mode on the watch?
> 
Yes it does

> And with “Enable Bedtime mode on DND sync” turned off, and “Enable only Bedtime mode on watch (No DND)” also turned off and with the other two settings enabled, are you able to sync DND normally?
>
Yes DND sync still works normally like this

Only thing missing is bedtime detection on Samsung, once we get that in, it'll be perfect.

**Comment by dreadedlama** [comment]
Great, thanks for confirmation @michael-r-elp. I'll proceed with merging the pixel fix in the master branch in that case. 

>Only thing missing is bedtime detection on Samsung, once we get that in, it'll be perfect.

Sure, looking into that. facing few hiccups. Will share the test build once done.

**Comment by Dave247** [comment]
> Oh, I forgot to mention that I haven’t added the Samsung fix yet.
> 
> So, if possible, could you validate the Pixel Watch fix using a non-Samsung phone. This would help us confirm that the fix is specific to the watch-side behaviour and isn’t dependent on Samsung specific functionality.

@dreadedlama Unfortunately I don't currently have access to any non Pixel or Samsung phones. Even close family all have Pixel phones (thanks to my hand-me-downs). Though there is some older devices that are sitting unused with a family member I was thinking to take back to try putting a more updated custom rom on which I think _should_ then work with digital wellbeing, but I can't say when I will get around to that. Though it looks like thankfully michael was able to test it for you.

**Comment by dreadedlama** [comment]
@Dave247 No worries, and thanks to you, @michael-r-elp, and everyone else who helped and tested along the way. I'm glad we were finally able to fix the Pixel Watch issue after all these years!
I'm working on the Samsung mobile side now, so hopefully we can get that sorted soon too.

**Comment by dreadedlama** [comment]
Fixed in **v3.1** 🎉

The Pixel Watch Bedtime Mode sync issue has been resolved. The fix is included in the [v3.1 release](https://github.com/dreadedlama/dnd-sync/releases/tag/v3.1).
Please update to v3.1 and let me know if you're still experiencing the issue.

I'll post an update on the Samsung mobile issue once I make some progress.

**Comment by michael-r-elp** [comment]
> Fixed in **v3.1** 🎉
> 
> The Pixel Watch Bedtime Mode sync issue has been resolved. The fix is included in the [v3.1 release](https://github.com/dreadedlama/dnd-sync/releases/tag/v3.1).
> Please update to v3.1 and let me know if you're still experiencing the issue.
> 
> I'll post an update on the Samsung mobile issue once I make some progress.
> 

Great, thanks for the work you put in, we might want to open up a different issue to track the Samsung bedtime sync, since this one has become quite a long conversation and was about the pixel sync issue anyway, so it might make sense to close this one then
